lib/ctbl.gi

```diff
@@ -2185,70 +2204,64 @@ InstallMethod( ClassPositionsOfSolvableRadical,
-##  by $K_1 = G$, and $K_{i+1} = [ K_i, G ]$.
-##  `LowerCentralSeries( <tbl> )' is a list
-##  $[ C_1, C_2, \ldots, C_n ]$ where $C_i$ is the set of positions of
-##  $G$-conjugacy classes contained in $K_i$.
-##
-##  Given an element $x$ of $G$, then $g\in G$ is conjugate to $[x,y]$ for
-##  an element $y\in G$ if and only if
-##  $\sum_{\chi\in Irr(G)} \frac{|\chi(x)|^2 \overline{\chi(g)}}{\chi(1)}
-##  \not= 0$, or equivalently, if the structure constant
-##  $a_{x,\overline{x},g}$ is nonzero.
-##
-##  Thus $K_{i+1}$ consists of all classes $Cl(g)$ in $K_i$ for that there
-##  is an $x\in K_i$ such that $a_{x,\overline{x},g}$ is nonzero.
+##  Let <A>tbl</A> be the character table of the group <M>G</M>, say.
+##  The lower central series <M>[ K_1, K_2, \ldots, K_n ]</M> of <M>G</M>
+##  is defined by <M>K_1 = G</M>, and <M>K_{i+1} = [ K_i, G ]</M>.
+##  <C>ClassPositionsOfLowerCentralSeries( <A>tbl</A> )</C> is a list
+##  <M>[ C_1, C_2, \ldots, C_n ]</M> where <M>C_i</M> is the set of positions
+##  of <M>G</M>-conjugacy classes contained in <M>K_i</M>.
+##  <P/>
+##  Given an element <M>x \in K_i</M>, <M>g^{-1} \in G</M> is conjugate to
+##  <M>[x,y]</M> for an element <M>y \in G</M> if and only if
+##  <M>\sum_{{\chi \in Irr(G)}} |\chi(x)|^2 \chi(g) / \chi(1) \neq 0</M>,
+##  see&nbsp;<Cite Key="Isa76" Where="Problem 3.10"/>,
+##  or equivalently, if the structure constant <M>a_{g, x, x}</M> is nonzero.
+##  <P/>
+##  Thus <M>K_{i+1}</M> consists of the normal closure in <M>G</M>
+##  of all classes <M>g^G</M> inside <M>K_i</M> for which there
+##  is an <M>x \in K_i</M> such that <M>a_{g, x, x}</M> is nonzero.
 ##
 InstallMethod( ClassPositionsOfLowerCentralSeries,
     "for an ordinary table",
     [ IsOrdinaryTable ],
     function( tbl )
     local series,     # list of normal subgroups, result
           K,          # actual last element of `series'
-          inv,        # list of inverses of classes of `tbl'
           mat,        # matrix of structure constants
           i, j,       # loop over `mat'
-          running,    # loop not yet terminated
           new;        # next element in `series'
 
-    series:= [];
-    series[1]:= [ 1 .. NrConjugacyClasses( tbl ) ];
+    series:= [ [ 1 .. NrConjugacyClasses( tbl ) ] ];
     K:= ClassPositionsOfDerivedSubgroup( tbl );
     if K = series[1] then
       return series;
     fi;
     series[2]:= K;
 
-    # Compute the structure constants $a_{x,\overline{x},g}$ with $g$ and $x$
-    # in $K_2$.
+    # Compute the structure constants $a_{g,x,x}$ with $g$ and $x$ in $K_2$.
     # Put them into a matrix, the rows indexed by $g$, the columns by $x$.
-    inv:= InverseClasses( tbl );
     mat:= List( K, x -> [] );
     for i in [ 2 .. Length( K ) ] do
       for j in K do
-        mat[i][j]:= ClassMultiplicationCoefficient( tbl, K[i], j, inv[j] );
+        mat[i,j]:= ClassMultiplicationCoefficient( tbl, K[i], j, j );
```

Comment:
I have no idea whether this is correct or not, so I'll just trust that you know what you are doing :-).